app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -7,13 +7,90 @@ package org.mozilla.focus.tiles
 import android.content.Context
 import android.content.Context.MODE_PRIVATE
 import android.content.SharedPreferences
+import android.graphics.BitmapFactory
 import android.support.annotation.UiThread
 import org.json.JSONArray
+import org.json.JSONObject
+import org.mozilla.focus.utils.UrlUtils
+import java.net.URL
 import java.util.UUID
 
 private const val PREF_HOME_TILES = "homeTiles"
 private const val CUSTOM_SITES_LIST = "customSitesList"
 
+private const val BUNDLED_HOME_TILES_DIR = "bundled/"
+private const val HOME_TILES_JSON_PATH = BUNDLED_HOME_TILES_DIR + "bundled_tiles.json"
+private const val HOME_TILES_JSON_KEY = "bundled_tiles"
+
+class BundledTilesManager private constructor(context: Context) {
+    companion object {
+        private var thisInstance: BundledTilesManager? = null
+        fun getInstance(context: Context): BundledTilesManager {
+            if (thisInstance == null) {
+                thisInstance = BundledTilesManager(context)
+            }
+            return thisInstance!!
+        }
+    }
+
+    private var bundledTilesCache = loadBundledTilesCache(context)
+
+    private fun loadBundledTilesCache(context: Context): LinkedHashMap<URL, BundledHomeTile> {
+        val tilesJSONString = context.assets.open(HOME_TILES_JSON_PATH).bufferedReader().use { it.readText() }
+        val tilesJSONArray = JSONObject(tilesJSONString).getJSONArray(HOME_TILES_JSON_KEY)
+        val lhm = LinkedHashMap<URL, BundledHomeTile>()
+        for (i in 0 until tilesJSONArray.length()) {
+            val tile = BundledHomeTile.fromJSONObject(tilesJSONArray.getJSONObject(i))
+            // TODO: Check for blacklisted sites and don't add them
+            lhm.put(URL(tile.url), tile)
+        }
+        return lhm
+    }
+
+    @UiThread
+    fun isURLPinned(urlString: String): Boolean {
+        val testUrl = URL(urlString)
+        for (u in bundledTilesCache.keys) {
+            if (compareUrl(testUrl, u)) return true
+        }
+        return false
+    }
+
+    private fun compareUrl(url1: URL, url2: URL): Boolean {
+        return url1.protocol == url2.protocol
```

Comment:
`.equals` won't work because we also check for things like mobile site vs non-mobile site. I just chose a bunch of these, I can add query too (but it won't ever be in any of our default sites, I'm thinking).

Reply:
The google video site uses a query param.